aspnetcore/host-and-deploy/docker/visual-studio-tools-for-docker.md

```diff
@@ -46,53 +48,119 @@ If the target framework is .NET Core, the **OS** drop-down allows for the select
 The Visual Studio Tools for Docker don't support adding Docker to an existing ASP.NET Core project targeting .NET Framework. For ASP.NET Core projects targeting .NET Core, there are two options for adding Docker support via the tooling. Open the project in Visual Studio, and choose one of the following options:
 
 * Select **Docker Support** from the **Project** menu.
-* Right-click the project in Solution Explorer and select **Add** > **Docker Support**.
+* Right-click the project in **Solution Explorer** and select **Add** > **Docker Support**.
+
+## Dockerfile overview
+
+A *Dockerfile*, the recipe for creating a final Docker image, is added to the project root. Refer to [Dockerfile reference](https://docs.docker.com/engine/reference/builder/) for an understanding of the commands within it. This particular *Dockerfile* uses a [multi-stage build](https://docs.docker.com/engine/userguide/eng-image/multistage-build/) with four distinct, named build stages:
+
+::: moniker range=">= aspnetcore-2.1"
+
+[!code-dockerfile[](visual-studio-tools-for-docker/samples/2.1/HelloDockerTools/Dockerfile.original?highlight=1,6,14,17)]
```

Comment:
Note in passing ... I'm glad to see you do this. I've wondered if we should adopt (where needed and with restricted use) code languages that aren't OOB from DocFX.

Reply:
I saw this is one of @JRAlexander's Docker docs and decided to carry it forward.